apps/service_discovery_http/src/sdh_socket.erl

```diff
@@ -0,0 +1,62 @@
+-module(sdh_socket).
+
+-behaviour(gen_server).
+
+-export([start_link/2]).
+
+-export([init/1,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         code_change/3,
+         terminate/2]).
+
+%% public api
+
+start_link(ListenOpts, AcceptorOpts) ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [ListenOpts, AcceptorOpts], []).
+
+%% gen_server api
+
+init([ListenOpts, AcceptorOpts]) ->
+    Port = maps:get(port, ListenOpts, 3000),
+    IPAddress = maps:get(ip, ListenOpts, {0,0,0,0}),
+    AcceptorPoolSize = maps:get(pool_size, AcceptorOpts, 10),
+    SocketOpts = maps:get(socket_opts, ListenOpts, [{reuseaddr, true},
+                                                    {nodelay, true},
+                                                    {reuseaddr, true},
+                                                    {backlog, 32768},
+                                                    {keepalive, true}]),
+    % Trapping exit so can close socket in terminate/2
+    _ = process_flag(trap_exit, true),
+    Opts = [{active, false}, {mode, binary}, {packet, raw}, {ip, IPAddress} | SocketOpts],
+    case gen_tcp:listen(Port, Opts) of
+        {ok, Socket} ->
+            MRef = monitor(port, Socket),
+            sdh_pool:accept_socket(Socket, AcceptorPoolSize),
```

Comment:
sdh_pool:start_link is never called though? Not sure about the code structure here.

Reply:
It is https://github.com/adoptingerlang/service_discovery/pull/9/files#diff-14c46a87868eb6f2e736b568654a9026R43